homeassistant/components/sensor/qnap.py

```diff
@@ -352,7 +352,7 @@ def state(self):
             return data['health']
 
         if self.var_id == 'drive_temp':
-            return int(data['temp_c'])
+            return int(data['temp_c']) if data['temp_c'] is not None else 0
```

Comment:
If the error is KeyError (as you describe), then this line is simpler:

return int(data['temp_c']) if 'temp_c' in data else 0

Reply:
LGTM